--- a/src/mktplinkfw2.c	2026-05-17 20:24:32.673909654 +0800
+++ a/src/mktplinkfw2.c	2026-05-17 20:27:49.986627206 +0800
@@ -144,6 +144,12 @@
 		.kernel_ep	= 0x800061b0,
 		.rootfs_ofs	= 0x140000,
 	}, {
+		.id		= "32Mltq",
+		.fw_max_len	= 0x1f90000,
+		.kernel_la	= 0x80002000,
+		.kernel_ep	= 0x800061b0,
+		.rootfs_ofs	= 0x140000,
+	}, {
 		.id		= "8Mmtk",
 		.fw_max_len	= 0x7a0000,
 		.kernel_la	= 0x80000000,
